File: packages/shared/utils/event.ts
import { v4 as uuidv4 } from "uuid";
import type { EventEnvelope } from "../types/event";

export function createEvent<T>(
  type: string,
  source: string,
  version: string,
  payload: T,
): EventEnvelope<T> {
  return {
    id: uuidv4(),
    type: type,
    timestamp: new Date(),
    payload: payload,
    source: source,
    version: version,
  };
}

File: apps/backend/src/shared/utils/event.py
from uuid import uuid4
from datetime import datetime, timezone
from typing import TypeVar

from src.shared.types.event import EventEnvelope

T = TypeVar("T")


def create_event(
    type: str,
    source: str,
    version: str,
    payload: T,
) -> EventEnvelope[T]:
    return EventEnvelope(
        id=uuid4(),
        type=type,
        timestamp=datetime.now(timezone.utc),
        payload=payload,
        source=source,
        version=version,
    )
